operator/internal/status/lokistack.go

```diff
@@ -113,18 +123,26 @@ func generateCondition(ctx context.Context, cs *lokiv1.LokiStackComponentStatus,
 	return conditionReady, nil
 }
 
-func checkForZoneawareNodes(ctx context.Context, k client.Client, zones []lokiv1.ZoneSpec) (bool, error) {
+func checkForZoneawareNodes(ctx context.Context, k client.Client, zones []lokiv1.ZoneSpec) (bool, bool, error) {
 	nodeLabels := client.HasLabels{}
 	for _, z := range zones {
 		nodeLabels = append(nodeLabels, z.TopologyKey)
 	}
 
 	nodeList := &corev1.NodeList{}
 	if err := k.List(ctx, nodeList, nodeLabels); err != nil {
-		return false, err
+		return false, false, err
+	}
+
+	for _, node := range nodeList.Items {
+		for _, nodeLabel := range nodeLabels {
+			if node.Labels[nodeLabel] == "" {
+				return true, false, nil
+			}
+		}
 	}
 
-	return len(nodeList.Items) > 0, nil
+	return len(nodeList.Items) > 0, true, nil
```

Comment:
I think I'd move the "are there nodes in the list?" check above the check for labels:

```suggestion
	if len(nodeList.Items) == 0 {
		return false, false, nil
	}

	for _, node := range nodeList.Items {
		for _, nodeLabel := range nodeLabels {
			if node.Labels[nodeLabel] == "" {
				return true, false, nil
			}
		}
	}

	return true, true, nil
```